app/components/markdown-block.js

```diff
@@ -4,40 +4,213 @@ import { createUseStyles } from 'react-jss'
+  //         {textSection}
+          
+  //       </div> 
+  //       </div>
+  //     </div>
+  //   );
+  // }
+
+   if (imgUrl && (imgSide === 'left' || imgSide === 'right')) {
+    return (
+      <div className={cx(classes.markdownBlock, classes[mode], className)} {...otherProps}>
+        <div className={classes.wrapperWithImage}>
+          {imgSide === 'left' && imageComponent}
+          {textSection}
+          {imgSide === 'right' && imageComponent}
+        </div>
+      </div>
+    );
+  }
+  
+  console.log("blocks", blockarray);
+  if (blockarray.length > 1) {
   return (
+    <div className={cx(classes.markdownBlock, classes[mode], className)} {...otherProps}>
+      {blockarray.map((block, index) => (
+        <div key={index} className={cx(classes.wrapper, block.imgSide === 'top' && classes.topLayout)}>
+          {block.imgUrl && block.imgSide === 'top' && (
+            <>
+              {/* Image Component */}
+              <img className={classes.imageStyleTop} src={block.imgUrl} alt="Markdown Block" />
+              
+              {/* Text Section */}
+              <div className={classes.textSectionWithImageTop}>
+                <MarkDown className={classes.mdclasses} options={{ overrides: { ActionButton: { component: ActionButton } } }}>
+                  {block.children}
+                </MarkDown>
+              </div>
+            </>
+          )}
+        </div>
+      ))}
+    </div>
+  );
+}
+ 
+  console.log("blocks1", blocks);
+  
+  
+   return (
     <div className={cx(classes.markdownBlock, classes[mode], className)} {...otherProps}>
       <div className={classes.wrapper}>
-        <MarkDown className={classes.mdclasses} options={{ overrides: { ActionButton: { component: ActionButton } } }}>
-          {children}
-        </MarkDown>
+        {textSection}
       </div>
     </div>
-  )
-}
+  );
+
+ 
+
+};
+
 export default TextBlock
 
 const useStylesFromThemeFunction = createUseStyles(theme => ({
   markdownBlock: {
+    display: 'flex',
+    flexDirection: props => (props.imgSide === 'top' ? 'row' : 'row'),
     textAlign: 'center',
     paddingTop: '1rem',
     paddingBottom: '4rem',
   },
   wrapper: {
+     display: 'flex',
+     flexDirection: props => (props.imgSide === 'top' ? 'column' : 'column'),
     maxWidth: theme.maxPanelWidth,
     marginLeft: 'auto',
     marginRight: 'auto',
     whiteSpace: 'pre-line',
   },
+  wrapperWithImage: {
+    display: 'flex',
+    flexDirection: (props) => (props.imgSide === 'top' ? 'row' : 'row'),
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    [`@media (max-width: ${theme.condensedWidthBreakPoint})`]: {
+      flexDirection: (props) => (props.imgSide === 'top' ? 'row' : 'column'),
+    },
+  },
+  
+  imageWrapper: {
+    display: 'inline-block',
+    textAlign: 'center',
+    marginBottom: '1rem', 
+  },
+  imageStyle: {
+
+    width: props => (props.imgSide === 'top' ? '9.375rem' : '40%'),  
```

Comment:
setting width (and height below) to 9.375rem will probably not work in all cases.